src/main/java/com/edmunds/tools/databricks/maven/UpsertClusterMojo.java

```diff
@@ -142,64 +157,79 @@ private void editCluster(ClusterTemplateModel ct, String clusterId) throws IOExc
                 .build();
 
         getDatabricksServiceFactory().getClusterService().edit(request);
-
     }
 
     private void startCluster(ClusterTemplateModel ct, String clusterId) throws IOException, DatabricksRestException, InterruptedException {
-        getLog().info(String.format("Starting cluster: name=[%s], id=[%s]", ct.getClusterName(), clusterId));
         ClusterService clusterService = getDatabricksServiceFactory().getClusterService();
         ClusterStateDTO clusterState = clusterService.getInfo(clusterId).getState();
-        if (clusterState == ClusterStateDTO.TERMINATED || clusterState == ClusterStateDTO.TERMINATING) {
-            clusterService.start(clusterId);
-        }
-        while (clusterService.getInfo(clusterId).getState() != ClusterStateDTO.RUNNING) {
-            Thread.sleep(5000);
+        if (clusterState != ClusterStateDTO.RUNNING) {
+            getLog().info(String.format("Starting cluster: name=[%s], id=[%s]. Current state=[%s]", ct.getClusterName(), clusterId, clusterState));
+            if (clusterState == ClusterStateDTO.TERMINATED || clusterState == ClusterStateDTO.TERMINATING
+                    || clusterState == ClusterStateDTO.ERROR || clusterState == ClusterStateDTO.UNKNOWN) {
+                clusterService.start(clusterId);
+            }
+            while (clusterService.getInfo(clusterId).getState() != ClusterStateDTO.RUNNING) {
+                Thread.sleep(5000);
```

Comment:
I think it would be useful to log something here.

Reply:
Done.